linked_list/main.py

```diff
@@ -62,34 +67,72 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
-
+        self.start = None
+        self.end = None
+        self.length= 0 
+        if elements:
+            for elem in elements:
+                self.append(elem)
+                
     def __str__(self):
-        pass
+        return "start: {}, end: {}".format(self.start.elem,self.end.elem)
 
     def __len__(self):
-        pass
+        return self.length
 
     def __iter__(self):
-        pass
+        return iter(self.start, self.end)
 
     def __getitem__(self, index):
-        pass
+        target = self.start
+        if index == 0:
+            return target
+        else:
+            for i in range(index):
+                target = target.next
+        return target
 
     def __add__(self, other):
-        pass
+        for elem in other:
+           self.append(elem)
+        return self
+        
```

Comment:
smart way of doing add and iadd

i struggled with this and heavily relied on lists, this is a much better way than what i did